provisioning/device/samples/solutions/BestPracticeSampleX509/BestPracticeSampleX509.cs

```diff
@@ -0,0 +1,533 @@
+﻿// Copyright (c) Microsoft. All rights reserved.
```

Comment:
This sample looks like a copy of the DeviceReconnectionSample, but I wonder if all the extra fluff is required to demonstrate how to authenticate with x509 to DPS and then hub? It inflates the size of this sample quite a bit. Is that desirable?

Seems to me we'd simply need to authenticate and then send some telemetry, and then include a link to the device reconnection sample to say that for a device app to be proper, it needs to include other concerns like client lifetime, connection status monitoring, etc.

Reply:
The purpose of combining two samples is to give a full solution. I thought it would be nicer if users can see the full solution instead of having to open another sample and combine them themselves. But I see your point.

Reply:
Perhaps we should have a team talk about it, but I feel this is too much. We don't do the same thing for every sample because it dilutes what a user should be learning by reading a sample. We want most of them focused on just 1 thing.